packages/db/src/models/pdr.ts

```diff
@@ -16,21 +17,28 @@ export default class PdrPgModel extends BasePgModel<PostgresPdr, PostgresPdrReco
     knexOrTrx: Knex | Knex.Transaction,
     pdr: PostgresPdr
   ) {
+    if (!pdr.created_at) {
+      throw new Error(`To upsert pdr record must have 'created_at' set: ${JSON.stringify(pdr)}`);
+    }
     if (pdr.status === 'running') {
       return knexOrTrx(this.tableName)
         .insert(pdr)
         .onConflict('name')
         .merge()
         // progress is not a required field, so trying to use `pdr.progress`
         // as where clause value throws a TS error
-        .where(knexOrTrx.raw('pdrs.execution_cumulus_id != EXCLUDED.execution_cumulus_id'))
-        .orWhere(knexOrTrx.raw('pdrs.progress < EXCLUDED.progress'))
+        .where(knexOrTrx.raw(`${this.tableName}.created_at <= to_timestamp(${translateDateToUTC(pdr.created_at)})`))
```

Comment:
This is probably tin-foil hat territory, but even though `tableName` comes from an internal file, I hesitated to interpolate it into a raw query string

Reply:
I tend to agree, however given the knex bug re: .where, this has to be a `.raw` query, and it's coming from a static value.    I'm open to alternate suggestions here.

Reply:
Can you refresh my memory on what the knex bug is re: where?

Reply:
I'll see if I can look it up in a moment, but using the 3 argument form of `.where` fails typescript compilation due to what I believe is a known bug in the typings for knex.

Reply:
OK. Not a big deal but I'd be interested in taking a quick look to see if there is any movement on that bug in Knex since it's very actively maintained

Reply:
Resolving since the next version of Knex has lots of breaking changes that we don't want to introduce now